ddio/file.cpp

```diff
@@ -156,3 +159,36 @@ void ddio_DoForeachFile(const std::filesystem::path &search_path, const std::reg
     }
   }
 }
+
+std::filesystem::path ddio_GetTmpFileName(const std::filesystem::path &basedir, const char *prefix) {
+  static const char alphanum[] =
+      "0123456789"
+      "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+      "abcdefghijklmnopqrstuvwxyz";
+  // size of random part
+  const int len = 10;
+  const char *ext = ".tmp";
+  std::filesystem::path result;
+  size_t len_result = strlen((basedir / prefix).u8string().c_str());
+  char *random_name = (char *)mem_malloc(len_result + len + strlen(ext) + 1);
+  strncpy(random_name, (basedir / prefix).u8string().c_str(), len_result);
+
+  srand(D3::ChronoTimer::GetTimeMS());
+
+  int tries = 20;
+  while (tries > 0) {
```

Comment:
I'm honestly not fond of this approach of retrying until we get a correct name (even if it's unlikely we have collisions). A better way to do it overall would be to use a hash (md5) of the file we mirror if we can; in my understanding these temp files correspond to unique uncompressed graphics, sounds resources, etc. What do you think of that?

Reply:
Well, collisions can occurs, especially with current Linux `ddio_GetTempFileName()`, if generate names too quickly. Luckily we don't have to do in real play, and there no such issues.

Hash based names has limited use in project. Literally only shared library extraction code can use this feature.

Reply:
ok I got it, all fine then!